packages/liveblocks-react/src/room.tsx

```diff
@@ -1992,17 +1999,33 @@ function useSubscribeToRoomThread(roomId: string) {
 
   return useCallback(
     (threadId: string) => {
-      // TODO: Optimistically create an inbox notification for the thread (if not already present),
-      //       createComment, createThread, updateRoomSubscriptionSettings etc, don't do this at the moment?
+      const subscribedAt = new Date();
+
+      const { store, onMutationFailure } = getRoomExtrasForClient(client);
+      const optimisticId = store.optimisticUpdates.add({
+        type: "subscribe-to-thread",
+        threadId,
+        subscribedAt,
+      });
 
       client[kInternal].httpClient.subscribeToThread({ roomId, threadId }).then(
         () => {
-          // TODO: Apply result of subscribeToThread
+          // TODO: subscribeToThread returns the thread, should it return the subscription (we could use it here instead of using subscribedAt)?
```

Comment:
The `c` endpoints for (un)subscribing currently return the thread (like the non-`c` ones), but the returning thread doesn't contain any info about the newly-created subscription so it can't be applied as truth. Currently I just use the optimistic date as truth if it's a success.